src/utils/robust.ts

```diff
@@ -1,54 +1,31 @@
-/*
-    Swallows (and reports) exceptions. Designed to wrap around modules at the "bootstrap" level.
-    For example "comments throwing an exception should not stop auto refresh"
+/**
+ * Swallows and reports exceptions. Designed to wrap around modules at the "bootstrap" level.
  */
-
-import { convertError, reportError } from './report-error';
+import { reportError } from './report-error';
 
 type ModuleFunction = () => void;
 type Module = [string, ModuleFunction];
 type Modules = Module[];
 
-const catchErrors = (fn: ModuleFunction): Error | undefined => {
-	let error: Error | undefined;
-
-	try {
-		fn();
-	} catch (e) {
-		error = convertError(e);
-	}
-
-	return error;
-};
-
-const logError = (
-	moduleName: string,
-	error: Error,
-	tags?: Record<string, string>,
-): void => {
-	window.console.warn('Caught error.', error.stack);
-	reportError(error, { module: moduleName, ...tags }, false);
-};
-
-const catchAndLogError = (
-	name: string,
-	fn: ModuleFunction,
-	tags?: Record<string, string>,
-): void => {
-	const error = catchErrors(fn);
-
-	if (error) {
-		logError(name, error, tags);
-	}
-};
-
-const catchErrorsWithContext = (
+const catchErrorsAndReport = (
```

Comment:
Refactored this module to combine all the functions and updated the test `robust.spec` accordingly